pkg/ipmi/ipmi.go

```diff
@@ -278,27 +268,44 @@ func (i *IPMI) RawSendRecv(msg Msg) ([]byte, error) {
 		msg:     recvMsg,
 	}
 
-	t := time.After(timeout)
-	for {
-		if err := ioctlGetRecv(i.Fd(), _IPMICTL_RECEIVE_MSG, recv); err != nil {
-			return nil, err
+	// This will be passed to the RawConn Read function. Until readMsg returns
+	// true, Read blocks and retries until the connection is ready for reading.
+	var result []byte
+	var rerr error
+	readMsg := func(fd uintptr) bool {
+		if err := ioctlGetRecv(fd, _IPMICTL_RECEIVE_MSG, recv); err != nil {
```

Comment:
most upstream callers seem to be using IPMICTL_RECEIVE_MSG_TRUNC, we should switch to it as well. (it means no backed up messages that we can't read)